src/test/java/edu/ucsb/cs156/gauchoride/controllers/DriverShiftControllerTests.java

```diff
@@ -0,0 +1,109 @@
+package edu.ucsb.cs156.gauchoride.controllers;
+
+import edu.ucsb.cs156.gauchoride.ControllerTestCase;
+import edu.ucsb.cs156.gauchoride.entities.User;
+import edu.ucsb.cs156.gauchoride.repositories.UserRepository;
+import edu.ucsb.cs156.gauchoride.entities.DriverShift;
+import edu.ucsb.cs156.gauchoride.repositories.DriverShiftRepository;
+import edu.ucsb.cs156.gauchoride.entities.DriverShift.Weekday;
+import edu.ucsb.cs156.gauchoride.testconfig.TestConfig;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Import;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MvcResult;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.csrf;
+
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.time.LocalTime;
+
+@WebMvcTest(controllers = DriverShiftController.class)
+@Import(TestConfig.class)
+public class DriverShiftControllerTests extends ControllerTestCase {
+    @MockBean
+    DriverShiftRepository driverShiftRepository;
+
+    @MockBean
+    UserRepository userRepository;
+
+    @Test
+    public void logged_out_users_cannot_get_all() throws Exception {
+        mockMvc.perform(get("/api/drivershifts/all"))
+                .andExpect(status().is(403));
+    }
+
+    @WithMockUser(roles = { "USER" })
+    @Test
+    public void logged_in_regular_user_cannot_get_all() throws Exception {
+        mockMvc.perform(get("/api/drivershifts/all")).andExpect(status().is(403));
+    }
+
+    @WithMockUser(roles = { "ADMIN", "DRIVER", "USER" })
+    @Test
+    public void logged_in_admin_or_driver_can_get_all_shifts() throws Exception {
+        mockMvc.perform(get("/api/drivershifts/all")).andExpect(status().is(200));
+    }
+
+    @WithMockUser(roles = { "ADMIN", "DRIVER", "USER" })
```

Comment:
Again: this needs to be separated into DRIVER and ADMIN if that's your intention; two differents tests, one with:

```java
@WithMockUser(roles = { "ADMIN", "USER" })
```

and another with:
```java
@WithMockUser(roles = { "DRIVER", "USER" })
```